src/controllers/auth.ts

```diff
@@ -36,13 +40,7 @@ const signup = (req: Request, res: Response, next: NextFunction): void => {
       message: 'Created successfully',
       data: [{ name, email, phone }],
     }))
-    .catch((err: CustomError | joiInterface) => {
-      if ('isJoi' in err) {
-        next(new CustomError(406, err.details[0].message));
-      } else {
-        next(new CustomError(500, 'server error'));
-      }
-    });
+    .catch((error)=>next(error));
```

Comment:
we should determine if the error came from validation or else
so make it like this 
.catch((err: CustomError | joiInterface) => {
      if ('isJoi' in err) {
        next(new CustomError(406, err.details[0].message));
      } else {
        next(err);
      }
    });